### PR TITLE
Layer as UIView in window

### PR DESCRIPTION
**Pull request by dima-avdeev-jb**
- Now new layer adds to viewController.window
 - Remove DensityProvider and simplify density usage
 - Also added scrim color and onOutsidePointerEvent listener

### CODE REVIEW

**Review by MatkovIvan — APPROVED** [review]

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/UIViewComposeSceneLayer.uikit.kt [hunk 1]
```diff
@@ -33,24 +33,53 @@ import androidx.compose.ui.window.ProvideContainerCompositionLocals
...
     configuration: ComposeUIViewControllerConfiguration,
     focusStack: FocusStack<UIView>?,
     windowInfo: WindowInfo,
     compositionContext: CompositionContext,
     compositionLocalContext: CompositionLocalContext?,
 ) : ComposeSceneLayer {
 
+    override var focusable: Boolean = focusStack != null
+    private var onOutsidePointerEvent: ((dismissRequest: Boolean) -> Unit)? = null
+    private val rootView = composeContainer.view.window ?: composeContainer.view
```
Comment (ASalavei) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/UIViewComposeSceneLayer.uikit.kt:59]: If composeContainer is detached from window, we're using its view, which is safety measure, but not the way how compose sub-layer is expected to work.
If we have control over `ComposeContainer`, we can detect when it become attached (and detached) in window and use this fact to attach(detach) other scene layers.

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/UIViewComposeSceneLayer.uikit.kt [hunk 2]
```diff
@@ -33,24 +33,53 @@ import androidx.compose.ui.window.ProvideContainerCompositionLocals
...
+            NSLayoutConstraint.activateConstraints(
+                getConstraintsToFillParent(this, rootView)
+            )
+        }
+
+        override fun pointInside(point: CValue<CGPoint>, withEvent: UIEvent?): Boolean {
+            //TODO pass invoke(true) on touch up event only when this touch event begins outside of layer bounds.
+            // Also it should be only one touch event (not multitouch with 2 and more touches).
+            // In other cases pass invoke(false)
+            onOutsidePointerEvent?.invoke(true)
```
Comment (ASalavei) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/UIViewComposeSceneLayer.uikit.kt:75]: What about?:
```
val pointInside = super. pointInside(point, withEvent)
if (!pointInside) {
    onOutsidePointerEvent?.invoke(true)
}
return pointInside
```

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/UIViewComposeSceneLayer.uikit.kt (hunk 1 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/UIViewComposeSceneLayer.uikit.kt:59]: In fact rootView is always presented now and it works like expected. Also I we will change this behaviour to use multiple UIWindow - this place we be cnahged again.
So, maybe let's stay as is for now?

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/UIViewComposeSceneLayer.uikit.kt (hunk 2 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/UIViewComposeSceneLayer.uikit.kt:75]: This is background view. So we always shoud call onOutsidePointerEvent. But in future we should add logic to handle different type of event like TOUCH_DOWN, TOUCH_UP.

**Review by elijah-semyonov — APPROVED** [review]

**Pull request merged**